habitat_baselines/motion_planning/mp_spaces.py

```diff
@@ -0,0 +1,220 @@
+import os
+import os.path as osp
+from abc import ABC, abstractmethod
+
+import numpy as np
+from PIL import Image
+
+from habitat_baselines.motion_planning.robot_target import RobotTarget
+
+try:
+    from ompl import base as ob  # pylint: disable=import-error
+    from ompl import geometric as og  # pylint: disable=import-error
+except ImportError:
+    pass
+
+
+def to_ob_state(vec, space, dim):
+    ob_vec = ob.State(space)
+    for i in range(dim):
+        ob_vec[i] = vec[i]
+    return ob_vec
+
+
+class MpSpace(ABC):
+    """
+    Defines an abstract planning space for OMPL to interface with.
+    """
+
+    def __init__(self, use_sim, ik):
+        self._mp_sim = use_sim
+        self._ik = ik
+
+    @abstractmethod
+    def convert_state(self, x):
+        pass
+
+    @abstractmethod
+    def set_arm(self, x):
+        pass
+
+    def set_env_state(self, env_state):
+        self.env_state = env_state
+
+    @abstractmethod
+    def get_range(self):
+        pass
+
+    @abstractmethod
+    def get_state_lims(self, restrictive=False):
+        pass
+
+    @abstractmethod
+    def get_state_dim(self):
+        pass
+
+    @abstractmethod
+    def convert_sol(self, path):
+        pass
+
+    @abstractmethod
+    def get_planner(self, si):
+        pass
+
+    @abstractmethod
+    def set_problem(
+        self, pdef, space, si, start_state, targ_state, is_state_valid
+    ):
+        pass
+
+    def render_start_targ(self, render_dir, subdir, targ_state):
+        pass
+
+
+def getPathLengthObjWithCostToGo(si):
+    obj = ob.PathLengthOptimizationObjective(si)
+    obj.setCostToGoHeuristic(ob.CostToGoHeuristic(ob.goalRegionCostToGo))
+    return obj
+
+
+class JsMpSpace(MpSpace):
+    def __init__(self, use_sim, ik, start_num_calls, should_render):
+        super().__init__(use_sim, ik)
+        # self._lower_joint_lims, self._upper_joint_lims = self._ik.get_joint_limits()
+        joint_lims = self.get_state_lims(True)
+        self._lower_joint_lims, self._upper_joint_lims = (
+            joint_lims[:, 0],
+            joint_lims[:, 1],
+        )
+        self.num_calls = start_num_calls
+        self._should_render = should_render
+
+    def convert_state(self, x):
+        return np.array([x[i] for i in range(7)])
```

Comment:
Type here? If x is an array the list comprehension is unnecesary cn be used as the getter into x.

Reply:
The type is some OMPL internal state vector. I am not sure the proper way to convert it to numpy array.